### PR TITLE
feat(data-exploration): bugfixes and improvements

### PR DESCRIPTION
**Pull request by mariusandra**
## Problem

Data exploration is not over the line yet, which means it's not live and enabled for some users.

## Changes

Tries to take it over the line. In scope (might change):

- [x] fix persons search
- [x] add "full: true" to the nodes
- [x] replace EventsNode with EventsQuery everywhere, except insight nodes
- [x] restore load more 
- [x] restore load new
- [x] api request cancellation into `dataNodeLogic`
- [x] add query running time next to "reload" button (behind a flag)

It should again work stable enough that we can enable the flag for ourselves.

## Will get to in the next PR

Planned to, but didn't manage to get this in here.

- [ ] add a date filter on top of the events scene (date_from & date_to !== before & after, need some unfangling)
- [ ] add date in-between rows - fixes https://github.com/PostHog/posthog/issues/13223 -  but I'm still not convinced we should do it - you either have so many events the day boundary is invisible, or so few that you only see day boundaries every few lines... I'm not sure it makes a lot of sense for both cases to show this, but I'll make something. I'll probably be a hack into LemonTable, `renderInBetweenRowOrNull(prev, next)` or something like that
- [ ] convert cells to filters on click

## Out of scope for now

- [ ] Unfangling `query_event_list.py` and `event.py`
- [ ] Add "unique users count" aggregation, and the other new ones

### CODE REVIEW

**Review by pauldambra — APPROVED** [review]
A few thoughts but nothing blocking

(I didn't run this :D)

As discussed at standup really we need _at least_ E2E tests on this. Would be good for the storybook section to have a "how do I start using data exploration" page too

File: frontend/src/queries/examples.ts [hunk 1]
```diff
@@ -74,11 +71,13 @@ const PropertyFormulas: EventsQuery = {
 
 const PropertyFormulasTable: DataTableNode = {
     kind: NodeKind.DataTableNode,
+    full: true,
```
[the remark this answers is not in the document]
Comment (mariusandra) [reply frontend/src/queries/examples.ts:74]: Normally I'd agree, but in this case I'm hoping we can convert this into a common notation that's easy to use everywhere.

E.g. `{ kind: 'DataTableNode', full: true, source: {} }`, `{ kind: 'SessionRecording', full: true, id: 'http://' }`, `{ kind: 'InsightViz', full: true, source: {} }`

If you pass `full: true`, you get the full scene with our _curated_ set of widgets and tools. Without `full`, we just display the element (recording, graph, etc).

I wanted it to be different from the other `showOneThing` keywords. I thought about "Chrome", but 1) it's a browser sadly and I do know where its name comes from, 2) some of the `OtherThings` in `showOtherThings` are not "chrome".

When in doubt, the simpler solution wins, so `full` it was 😅

**Review by thmsobrmlr — APPROVED** [review]

File: frontend/src/queries/nodes/DataNode/dataNodeLogic.ts [hunk 2]
```diff
@@ -25,52 +27,82 @@ export const dataNodeLogic = kea<dataNodeLogicType>([
...
     }),
-    loaders(({ actions, values }) => ({
+    loaders(({ actions, cache, values, props }) => ({
         response: [
             null as DataNode['response'] | null,
             {
-                loadData: async () => {
-                    return (await query<DataNode>(values.query)) ?? null
+                loadData: async (_, breakpoint) => {
+                    if (cache.abortController) {
```
[the remark this answers is not in the document]
Comment (mariusandra) [reply frontend/src/queries/nodes/DataNode/dataNodeLogic.ts]: I added an `abortQuery` action

File: frontend/src/queries/nodes/DataNode/dataNodeLogic.ts [hunk 3]
```diff
@@ -25,52 +27,82 @@ export const dataNodeLogic = kea<dataNodeLogicType>([
...
+                    try {
+                        const now = performance.now()
+                        const data = (await query<DataNode>(props.query, methodOptions)) ?? null
+                        breakpoint()
+                        actions.setElapsedTime(performance.now() - now)
+                        return data
+                    } catch (e: any) {
+                        cache.abortController = null
+                        if (e.name === 'AbortError' || e.message?.name === 'AbortError') {
+                            return values.response
```
[the remark this answers is not in the document]
Comment (mariusandra) [reply frontend/src/queries/nodes/DataNode/dataNodeLogic.ts:69]: happy to do this in a followup 😅

File: frontend/src/queries/nodes/DataNode/dataNodeLogic.ts [hunk 4]
```diff
@@ -152,9 +237,12 @@ export const dataNodeLogic = kea<dataNodeLogicType>([
     afterMount(({ actions }) => {
         actions.loadData()
     }),
-    beforeUnmount(({ actions, values }) => {
+    beforeUnmount(({ actions, cache, values }) => {
         if (values.autoLoadRunning) {
             actions.stopAutoLoad()
         }
+        if (cache.abortController) {
```
[the remark this answers is not in the document]
Comment (mariusandra) [reply frontend/src/queries/nodes/DataNode/dataNodeLogic.ts]: hmm... makes sense, though if all data goes through this dataNodeLogic, then we'll have cases where cancelling this before unmount will pay off... So I'd keep it. I changed it a bit to go through a new `abortQuery` action

**Comment by mariusandra** [comment]
Thanks for the reviews! I cleaned up a few things, and already had to merge with master. Since this PR lightly touches a lot of files (most of which are behind a flag), and fixes a few bad bugs, I'd happily merge this in and volunteer for oncall support for it over the next days 😅. Otherwise I fear work on other data exploration tasks next week will yield in a lot of annoying merge conflicts...

**Pull request merged**